### ISSUE TITLE
Very slow with lots of fields to render

### ISSUE THREAD

**Issue by Jimmylet** [Laravel-Backpack/CRUD#1153]
Hello all,

I have 3200 fields to render on a single create / edit view. Many checkboxes and radio buttons. It's very slow to load, more than 30 seconds. 

I know this is not a "normal" use but would I know if there is a way to avoid all this loading time?

I use tabs to separate my content, would not it be possible to load the content of other "tabs" at the clic? Or is there a way to improve performance?

Fields are added manually via the crud controller.

My debug bar say : 3191 templates were rendered.

### Backpack, Laravel, PHP, DB version:
```php
"require": {
        "php": ">=7.0.0",
        "backpack/base": "^0.8.2",
        "backpack/crud": "^3.3",
```

**Comment by tabacitu** [comment]
Whoow. I agree, the problem should be with loading a separate view for each field, so in total 3200 views, which is a lot. 

Backpack really wasn't created with more than a few hundred fields in mind, so I don't think a general solution is or will be available. But if most of your fields are a certain field type (let's say, ```text```) you could use a custom ```list.blade.php``` view with ```$this->crud->setListView('your-view');```. In that view, you can hardcode a ```switch``` statement for the most common field types you're using, and load the code right there, instead of including another view. If you have 2-3 fields that repeat 3000 times, this should drop the number of loaded views significantly. It's a hack, but... I don't think you have many options...

Hope it helps. Cheers!

**Comment by Jimmylet** [comment]
Thanks for you reply Tabacity. Yes, it's a lot of radio fields.

**Comment by tabacitu** [comment]
Oh sure. I meant a regular PHP switch statement:
```php
    switch (variable) {
        case 'value':
            # code...
            break;
        
        default:
            # code...
            break;
    }
```

That being said, I mentioned using a custom ```list.blade.php``` file. I meant ```create``` or ```update```, obviously, since that's the operation you're concerned about. Sorry. But since both of them are actually loading the fields inside ```form_content.blade.php```, that's the file you should be looking to edit.

Starting with line 16 your ```form_content.blade.php``` might look something like:
```diff
{{-- Show the inputs --}}
  @foreach ($fields as $field)
    <!-- load the view from the application if it exists, otherwise load the one in the package -->
+	@if($field['type']=='radio')
+              {{-- copy-paste the radio.blade.php file here --}}
-	@elseif(view()->exists('vendor.backpack.crud.fields.'.$field['type']))
-	@if(view()->exists('vendor.backpack.crud.fields.'.$field['type']))
		@include('vendor.backpack.crud.fields.'.$field['type'], array('field' => $field))
	@else
		@include('crud::fields.'.$field['type'], array('field' => $field))
	@endif
  @endforeach
```

This way, instead of including the radio button using yet another ```include```, you'll can be loading it from your ```form_content.blade.php``` file.

Hope it helps!

**Comment by Jimmylet** [comment]
@tabacitu thank you very much ! Your answer helped me solve my problem.

I included the contents of the radio field (`radio.blade.php`) in the file `show_fields.blade.php` as you explained.

I've also inserted all the includes contained in the file, for example, `<div @include ('crud :: inc.field_wrapper_attributes')>` or `crud::inc.field_translatable_icon `. I was able to go down to 500 includes on the page instead of 3200. It now loads in 8 seconds, which is more acceptable.

Thank you !
